packages/components/src/internal/query/filter.ts

```diff
@@ -109,28 +120,22 @@ export function getDateFieldLabKeySql(filter: Filter.IFilter): string {
             [startDateStart, startDateEnd] = getDateStrRange(filter.getValue());
         }
 
-        if (urlSuffix ===  Filter.Types.DATE_EQUAL.getURLSuffix()) {
-            return "(" + columnNameSelect + " >= " + startDateStart + " AND " + columnNameSelect + " < " + startDateEnd + ")";
-        } else if (urlSuffix ===  Filter.Types.DATE_NOT_EQUAL.getURLSuffix()) {
-            return "(" + columnNameSelect + " < " + startDateStart + " OR " + columnNameSelect + " >= " + startDateEnd + ")";
-        }
-        else if (urlSuffix ===  Filter.Types.BETWEEN.getURLSuffix()) {
-            return "(" + columnNameSelect + " >= " + startDateStart + " AND " + columnNameSelect + " < " + endDateEnd + ")";
-        }
-        else if (urlSuffix ===  Filter.Types.NOT_BETWEEN.getURLSuffix()) {
-            return "(" + columnNameSelect + " < " + startDateStart + " OR " + columnNameSelect + " >= " + endDateEnd + ")";
-        }
-        else if (urlSuffix ===  Filter.Types.DATE_GREATER_THAN.getURLSuffix()) {
-            return "(" + columnNameSelect + " >= " + startDateEnd + ")";
-        }
-        else if (urlSuffix ===  Filter.Types.DATE_LESS_THAN.getURLSuffix()) {
-            return "(" + columnNameSelect + " < " + startDateStart + ")";
-        }
-        else if (urlSuffix ===  Filter.Types.DATE_GREATER_THAN_OR_EQUAL.getURLSuffix()) {
-            return "(" + columnNameSelect + " >= " + startDateStart + ")";
-        }
-        else if (urlSuffix ===  Filter.Types.DATE_LESS_THAN_OR_EQUAL.getURLSuffix()) {
-            return "(" + columnNameSelect + " < " + startDateEnd + ")";
+        if (urlSuffix === Filter.Types.DATE_EQUAL.getURLSuffix()) {
+            return '(' + columnNameSelect + ' >= ' + startDateStart + ' AND ' + columnNameSelect + ' < ' + startDateEnd + ')';
+        } else if (urlSuffix === Filter.Types.DATE_NOT_EQUAL.getURLSuffix()) {
+            return '(' + columnNameSelect + ' < ' + startDateStart + ' OR ' + columnNameSelect + ' >= ' + startDateEnd + ')';
+        } else if (urlSuffix === Filter.Types.BETWEEN.getURLSuffix()) {
+            return '(' + columnNameSelect + ' >= ' + startDateStart + ' AND ' + columnNameSelect + ' < ' + endDateEnd + ')';
+        } else if (urlSuffix === Filter.Types.NOT_BETWEEN.getURLSuffix()) {
+            return '(' + columnNameSelect + ' < ' + startDateStart + ' OR ' + columnNameSelect + ' >= ' + endDateEnd + ')';
+        } else if (urlSuffix === Filter.Types.DATE_GREATER_THAN.getURLSuffix()) {
+            return '(' + columnNameSelect + ' >= ' + startDateEnd + ')';
```

Comment:
it's confusing here because startDateEnd actually is nextDateOfStartDate 00:00:00, so it's exclusive. I'll add a comment in getDateStrRange to clarify.